classes/article/ArticleDAO.inc.php

```diff
@@ -182,11 +183,14 @@ function updateObject($article) {
 		);
 
 		$this->updateLocaleFields($article);
+		$contextId = Request::getContext()->getId();
+		$version = $article->getCurrentVersionId($contextId);
 
 		// update authors for this article
-		$authors = $article->getAuthors();
+		$authors = $article->getAuthors(false, $version);
 		for ($i=0, $count=count($authors); $i < $count; $i++) {
 			if ($authors[$i]->getId() > 0) {
+				$authors[$i]->setVersion($version);
```

Comment:
Is it necessary to set the version here? -- because we search after the authors by the version (getAuthors (false, $version)?